### PR TITLE
Avl tree

### PR DESCRIPTION
**Pull request by horosahil**
## Related Issue
- Add code for AVL tree #301 

## Proposed Changes
- Added code for insertion, deletion and seraching in AVL tree inside B-BinarySearchTree folder which is inside T-Tree folder
    - file name : AVL.cpp
- Created readme.md inside B-BinarySearchTree folder and added all the necessary data regarding AVL tree in the readme.md file

## Additional Info
- Any additional information or context

## Checklist

- [x] ✅ My code follows the code style of this project.
- [x] 📝 My change requires a change to the README.
- [x] 🎀 I have updated the README accordingly.
- [x] 👀 I have read the [**CONTRIBUTING**](https://github.com/Lakhankumawat/LearnCPP/blob/main/Contributing.md) document.
- [x] ✨ I have added tests to cover my changes.
- [x] 🚩 All new and existing tests passed.
- [x] 🌟 ed the repo

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Output Screenshots
| Screenshot #1      |   
| ----------- | 
| Output of the AVL code | 
| output |

### CODE REVIEW

**Comment by Lakhankumawat** [comment]
///--------------------Pull Requests--------------------///

1. Short Description or CheckLists

Make sure your Pull Request is descriptive and complete the checklist .
I would recommend you to go through this tutorial video of the project [here](https://youtu.be/kIFwqp9B7pQ) ,thanks.


///------------------Description -----------------------///

For contribution and more guidelines i would recommend you to go through this 
tutorial video of the project [here](https://youtu.be/7UgC3HDETRA), thanks.



///------------------Description -----------------------///

Let me show you how to check the check list , there should be no space inside brackets like [x]
 `[x]`
- [x] Checklist
- [ ] Or you can just check them from outside without doing any of these
- [x] ✨ I have added tests to cover my changes.
- [x] 🚩 All new and existing tests passed.
- [ ] 🌟 ed the repo

1. Starr the repo if you haven't !
2. Have you watched this [video](https://youtu.be/kIFwqp9B7pQ) ?


///----------------------PR README --------------------///


You need to follow particular guidelines for your README , please follow
these guidelines [click here](https://github.com/Lakhankumawat/LearnCPP#to-add-programs-using-stl-follow-same-approach-as--algorithms) 
also you can see how your README should look like [see here](https://github.com/Lakhankumawat/LearnCPP/blob/main/S-SortingAlgorithms/README.md) 
for more please go through these [here](https://youtu.be/7UgC3HDETRA)

**Comment by horosahil** [comment]
Hi @Lakhankumawat 
I have completed all the require checklist and starred the repository. 
Anything else i have to do

File: T-Tree/B-BinarySearchTree/AVL.cpp [hunk 1]
```diff
@@ -0,0 +1,311 @@
...
+        bool val = search(root->right, key);
+        return val;
+    }
+}
+
+int main()
+{
+    Node *root = NULL;
+
+    root = insert(root, 9);
```
Comment (siddhi-244) [inline T-Tree/B-BinarySearchTree/AVL.cpp:254]: Can uh try to take this as input from the user ?

Comment (horosahil) [reply T-Tree/B-BinarySearchTree/AVL.cpp:254]: Yes i can try

Comment (horosahil) [reply T-Tree/B-BinarySearchTree/AVL.cpp]: I have made the changes in which user can input the data
output1

**Review by siddhi-244 — APPROVED** [review]
LGTM

**Review by Lakhankumawat — APPROVED** [review]
LGTM 💯

**Comment by Lakhankumawat** [comment]
@all-contributors please add @horosahil  code ,

**Pull request merged**